templates/demo/web3js-react-dapp-demo/src/AccountDetail.tsx

```diff
@@ -0,0 +1,283 @@
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+
+import { v4 as uuid } from "uuid";
+
+import type { SendTransactionEvents, NewHeadsSubscription } from "web3-eth";
+
+import { type IWeb3Context, Web3Context } from "./web3/Web3Context";
+import type {
+  DataFormat,
+  providers,
+  TransactionReceipt,
+  Web3PromiEvent,
+} from "web3";
+
+function AccountDetail({ address }: { address: string }) {
+  const web3Context: IWeb3Context = useContext(Web3Context);
+
+  const [transferTo, setTransferTo] = useState<string>("");
+  const [transferAmount, setTransferAmount] = useState<number>(NaN);
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+  const [transactions, setTransactions] = useState<Map<string, string>>(
+    new Map(),
+  );
+
+  type TransferEvent = Web3PromiEvent<
+    TransactionReceipt,
+    SendTransactionEvents<DataFormat>
+  >;
+  const transferEvents = useRef<Map<string, TransferEvent>>(new Map());
+
+  // update transactions
+  useEffect(() => {
+    const currentProvider: providers.EIP6963ProviderDetail | undefined =
+      web3Context.currentProvider;
+
+    if (currentProvider === undefined) {
+      return;
+    }
+
+    function clearTransactions() {
+      setTransactions(new Map());
+      transferEvents.current.forEach((transferEvent: TransferEvent) => {
+        transferEvent.removeAllListeners();
+      });
+      transferEvents.current = new Map();
+    }
+
+    currentProvider.provider.on("chainChanged", clearTransactions);
+
+    return () =>
+      currentProvider.provider.removeListener(
+        "chainChanged",
+        clearTransactions,
+      );
+  });
+
+  const [balance, setBalance] = useState<bigint | undefined>(undefined);
+  const subscriptionId: MutableRefObject<string | undefined> =
+    useRef(undefined);
+
+  // update balance
+  useEffect(() => {
+    updateBalance();
+
+    if (!web3Context.web3.subscriptionManager.supportsSubscriptions()) {
+      return;
+    }
+
+    web3Context.web3.eth
+      .subscribe("newBlockHeaders")
+      .then((newBlockSubscription: NewHeadsSubscription) => {
+        subscriptionId.current = newBlockSubscription.id;
+        newBlockSubscription.on("data", () => {
+          updateBalance();
+        });
+      });
+
+    return () => {
+      web3Context.web3.eth.subscriptionManager.unsubscribe(
+        ({ id }) => subscriptionId.current === id,
+      );
+    };
+  });
+
+  function updateBalance(): void {
+    web3Context.web3.eth.getBalance(address).then(setBalance);
+  }
+
+  function isValidAddress(address: string): boolean {
+    return /^(0x)?[0-9a-fA-F]{40}$/.test(address);
+  }
+
+  function transferFormChange(e: ChangeEvent<HTMLInputElement>): void {
+    let to: string = transferTo;
+    if (e.target.name === "to") {
+      to = e.target.value;
+      setTransferTo(to);
+    }
+
+    let amount: number = transferAmount;
+    if (e.target.name === "amount") {
+      amount = parseInt(e.target.value);
```

Comment:
amount should be a string, so we wouldn't need to parse. instead just make sure the amount is a valid number